### PR TITLE
Docs: Add smoke test examples to datasource guides

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

This PR adds examples on how to write basic smoke tests that verifies that data source plugin components loads. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md [hunk 1]
```diff
@@ -23,6 +23,18 @@ In many cases, the execution of annotation queries requires different handling t
 
 If your data source plugin implements a custom annotation editor, you can write tests that verify that the editor works as expected. If you haven't implemented a custom editor, then the plugin will use the built-in in editor. In that case, you don't need to write tests.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md:24]:
```suggestion
If your data source plugin implements a custom annotation editor, you can write tests that verify that the editor works as expected. If you haven't implemented a custom editor, then the plugin uses the built-in editor. In that case, it's not necessary to write tests.
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md [hunk 2]
```diff
@@ -23,6 +23,18 @@ In many cases, the execution of annotation queries requires different handling t
...
 
+The following example is a simple smoke test that verifies that a custom annotation editor loads.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md:26]:
```suggestion
The following example is a simple smoke test that verifies that a custom annotation editor loads:
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md [hunk 3]
```diff
@@ -23,6 +23,18 @@ In many cases, the execution of annotation queries requires different handling t
...
+test('should render annotations editor', async ({ annotationEditPage, page, readProvisionedDataSource }) => {
+  const ds = await readProvisionedDataSource({ fileName: 'datasources.yml' });
+  await annotationEditPage.datasource.set(ds.name);
+  await expect(page.getByRole('textbox', { name: 'Query Text' })).toBeVisible();
+});
+```
+
 ### Test the entire annotation query execution flow
 
 In the next example, we perform an integration test where we test a plugin's entire annotation query data flow.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/annotation-queries.md:40]:
```suggestion
In the following example, an integration test covers a plugin's entire annotation query data flow:
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md [hunk 4]
```diff
@@ -16,6 +16,20 @@ sidebar_position: 10
 
 Most data source plugins need authentication to communicate with third-party services. The appropriate place to configure the connection details is the data source configuration page, and the details there must be valid so that the health check endpoint used to test the configuration works as expected.
 
+### Test that the configuration editor loads
+
+The following example is a simple smoke test that verifies that the data source configuration editor loads.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md:21]:
```suggestion
The following example is a simple smoke test that verifies that the data source configuration editor loads:
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/custom-variable-editor.md [hunk 5]
```diff
@@ -17,7 +17,21 @@ sidebar_position: 30
 
 [Variable queries](https://grafana.com/docs/grafana/latest/dashboards/variables/add-template-variables/#add-a-query-variable) allow users to query a data source to load lists of data such as metric names. You can then reference variables in queries to make your dashboards more interactive and dynamic. If your data source plugin implements the custom variable support API, you may want to use the `variableEditPage` fixture to test that your plugin's variable implementation works as expected.
 
-### Test the custom variable editor
+### Test that the custom variable editor loads
+
+The following example is a simple smoke test that verifies that the custom variable editor loads.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/custom-variable-editor.md:22]:
```suggestion
The following example is a simple smoke test that verifies that the custom variable editor loads:
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md [hunk 6]
```diff
@@ -17,6 +17,20 @@ sidebar_position: 20
 
 The query editor is a central piece of a data source plugin as this is where users craft the query that is used to fetch data. Your data source plugin can provide a rich query editor which allows various query types that target different APIs. Your query editor can even have features such as visual query builders, IntelliSense, and autocompletion.
 
+### Test that the query editor loads
+
+The following example is a simple smoke test that verifies that the data source query editor loads.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md:22]:
```suggestion
The following example is a simple smoke test that verifies that the data source query editor loads:
```

**Review by josmperez — APPROVED** [review]
Looks good, just a few nits. LGTM.

**Review by academo — APPROVED** [review]

**Pull request merged**